src/ngraph/node.cpp

```diff
@@ -1102,3 +1104,45 @@ vector<Output<const Node>> Node::outputs() const
 
     return result;
 }
+
+bool Node::evaluate(const EvaluatorTensorVector& output_values,
+                    const EvaluatorTensorVector& input_values)
```

Comment:
I disagree with this signature. This function should accept a vector of inputs and returns (as a return value or as non-constant argument by a reference) a vector of outputs without any pre-requisites on calling validate_and_infer_types with constants. It does't look solid.

@diyessi if you don't have any quick options, please commit as-is we will find better solution (if any) later.